test/unittests/evm_test.cpp

```diff
@@ -777,35 +777,174 @@ TEST_F(evm, selfdestruct)
 
 TEST_F(evm, selfdestruct_with_balance)
```

Comment:
Should this test maybe be renamed to something like "selfdestruct_no_balance" since you're setting the account balance to 0?